components/camera/videosource/webcam.go

```diff
@@ -202,7 +204,33 @@ func NewWebcamSource(ctx context.Context, attrs *WebcamAttrs, logger golog.Logge
 	constraints := makeConstraints(attrs, debug, logger)
 
 	if attrs.Path != "" {
-		return tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		cam, err := tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		if err != nil {
+			return cam, err
+		}
+
+		goutils.PanicCapturingGo(func() {
+			src := camera.SourceFromCamera(cam)
+			defer func() {
+				if err := cam.Close(ctx); err != nil {
+					logger.Debugf("failed to close camera: %v", err)
+				}
+			}()
+			for {
+				if goutils.SelectContextOrWait(ctx, 500*time.Millisecond) {
+					// mediadevices connects to the OS to get the properties for a driver. If the OS no longer knows
+					// about a specific driver then properties will be empty, and we can safely assume the driver no
+					// longer exists and should be closed.
+					if len(gostream.PropertiesFromMediaSource[image.Image, prop.Video](src)) == 0 {
```

Comment:
For `gostream.PropertiesFromMediaSource[T, U any](gostream.MediaSource[T])` there's no way to derive `U` from `MediaSource[T]` and the compiler complains of this. Because of this I pass the type information explicitly.